### PR TITLE
dev-lang/janet: new package

### PR DESCRIPTION
**Pull request by oz123**
Janet is a functional and imperative programming language and bytecode
interpreter. It is a modern lisp, but lists are replaced by other data
structures with better utility and performance (arrays, tables, structs,
tuples). The language also supports bridging to native code written in C,
meta-programming with macros, and bytecode assembly.

Package-Manager: Portage-2.3.84, Repoman-2.3.16
Signed-off-by: Oz Tiram <oz.tiram@gmail.com>

### CODE REVIEW

File: dev-lang/janet/Manifest [hunk 1]
```diff
@@ -0,0 +1 @@
+DIST janet-1.6.0.tar.gz 956098 BLAKE2B 3555419ec24cedbd093f0b156702dbdf58cc9f0efcf52b87f49365eea69b1e10193d82b09d2574df4b5026e3f6e33edf14fc20eecc59f0106220f2b81c916bcb SHA512 9b4707a5121fdee887810b0f652c1acb4059ee6f802d9118c3fdc20421f660f0071d7833a2a6a5c929878d21216a34ee6af2ea190061c94b7b8585eca3d0f57a
```
Comment (juippis) [inline dev-lang/janet/Manifest]: 1.7.0 is available. Upstream also seems to have meson build option available, could you switch to use that instead?

File: dev-lang/janet/janet-1.6.0-r1.ebuild [hunk 2]
```diff
@@ -0,0 +1,47 @@
+# Copyright 2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+inherit flag-o-matic
```
Comment (juippis) [inline dev-lang/janet/janet-1.6.0-r1.ebuild]: One empty line below EAPI please. I didn't see flag-o-matic being used?

File: dev-lang/janet/janet-1.6.0-r1.ebuild [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+SRC_URI="https://github.com/janet-lang/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="static-libs"
+
+DEPEND=""
+RDEPEND="${DEPEND}"
+BDEPEND=""
```
Comment (juippis) [inline dev-lang/janet/janet-1.6.0-r1.ebuild]: These are pretty useless and can be removed.

File: dev-lang/janet/janet-1.6.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,38 @@
+# Copyright 2020 Gentoo Authors
```
Comment (juippis) [inline dev-lang/janet/janet-1.6.0.ebuild]: This ebuild seems useless as you have a newer one available.

File: dev-lang/janet/metadata.xml [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+<longdescription lang="en">
+Janet is a functional and imperative programming language and bytecode interpreter. It is a modern lisp, but lists are replaced by other data structures with better utility and performance (arrays, tables, structs, tuples). The language also supports bridging to native code written in C, meta-programming with macros, and bytecode assembly.
+</longdescription>
+<upstream>
+	<remote-id type="github">janet-lang/janet</remote-id>
+</upstream>
+<use>
+	<flag name="static-libs">install janet static library archine, you must enable this flag for using `jpm build`</flag>
+</use>
+</pkgmetadata>
```
Comment (juippis) [inline dev-lang/janet/metadata.xml]: Remember proper indentation.

File: dev-lang/janet/metadata.xml [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+<maintainer type="person">
+	<name>Oz Tiram</name>
+	<email>oz.tiram@gmail.com</email>
+</maintainer>
+<maintainer type="project">
+	<email>proxy-maint@gentoo.org</email>
+	<name>Proxy Maintainers</name>
+</maintainer>
+<longdescription lang="en">
+Janet is a functional and imperative programming language and bytecode interpreter. It is a modern lisp, but lists are replaced by other data structures with better utility and performance (arrays, tables, structs, tuples). The language also supports bridging to native code written in C, meta-programming with macros, and bytecode assembly.
```
Comment (juippis) [inline dev-lang/janet/metadata.xml]: Can be split to multiple lines and wrapped properly.

Comment (oz123) [reply dev-lang/janet/metadata.xml]: This is done in the new commit.

File: dev-lang/janet/janet-1.6.0.ebuild (hunk 4 above)
Comment (oz123) [reply dev-lang/janet/janet-1.6.0.ebuild]: I removed the ebuild for 1.6.0

File: dev-lang/janet/janet-1.6.0-r1.ebuild (hunk 3 above)
Comment (oz123) [reply dev-lang/janet/janet-1.6.0-r1.ebuild]: removed in 1.7.0.

File: dev-lang/janet/janet-1.6.0-r1.ebuild (hunk 2 above)
Comment (oz123) [reply dev-lang/janet/janet-1.6.0-r1.ebuild]: fixed.

File: dev-lang/janet/Manifest (hunk 1 above)
Comment (oz123) [reply dev-lang/janet/Manifest]: It's not officially supported, and I  am not familiar with meson ebuilds...

File: dev-lang/janet/metadata.xml (hunk 5 above)
Comment (oz123) [reply dev-lang/janet/metadata.xml]: What do you mean here?

Comment (oz123) [reply dev-lang/janet/metadata.xml]: This is done in the new commit.

File: dev-lang/janet/janet-1.7.0.ebuild [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+SRC_URI="https://github.com/janet-lang/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="static-libs"
+
+PATCHES=(
+	"${FILESDIR}/${P}"-fix-missing-soname.patch
+	)
```
Comment (juippis) [inline dev-lang/janet/janet-1.7.0.ebuild]: Extra indent.

Comment (oz123) [reply dev-lang/janet/janet-1.7.0.ebuild]: done.

**Comment by juippis** [comment]
> ```
>  * Messages for package dev-lang/janet-1.7.0:
> 
>  * QA Notice: Files built without respecting LDFLAGS have been detected
>  *  Please include the following list of files in your report:
>  * /usr/bin/janet
>  * /usr/lib64/libjanet.so
> ```
> 
> Looks good otherwise.

Not fixed yet.

**Pull request closed without merging**